### DISCUSSION TITLE
Maybe a good way for format?

### DISCUSSION THREAD

**Discussion by yinghaoyu** [ayamir/nvimdots#1147] [Ideas]
As cpp project, at most time I think that `.clang-format` in the project root path should be matched first. If it does not exist, the [default file](https://github.com/ayamir/nvimdots/blob/main/lua/modules/configs/completion/formatters/clang_format.lua) is matched.

After each upgrade period, It's irrational to turn `return { "-style={BasedOnStyle: LLVM, IndentWidth: 4}" }` to `return { "-style=file" }` for users.

**Comment by ayamir** [comment]
Create file `lua/user/configs/formatters/clang_format.lua` and fill it with `return { "-style=file" }`. We respect user's config first.

**Reply by yinghaoyu** [reply]
It make sense but not conivent for each upgrade. As you can see, **nvimdots** is an active repository.

**Reply by charliie-dev** [reply]
what upgrade? isn't that being add to gitignore already?

**Reply by yinghaoyu** [reply]
ohhhh, I should use `git pull` at `~/.config/nvim` rather than `./scripts/install.sh`.

**Reply by yinghaoyu** [reply]
Not working at Version `0.11` now.

**Reply by yinghaoyu** [reply]
@ayamir

**Reply by ayamir** [reply]
It works for me now. We removed `main` branch and use `0.11` as default branch. Maybe you should checkout to 0.11 branch mannually.
